lib/core/utils/get-tag-name.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns the tagName,
+ * if it is a HTMLElement it gets lowercased
+ * @param  {Element} node element
+ * @return {String}       normalized tagName
+ */
+axe.utils.getTagName = function(node) {
```

Comment:
It is / was mostly specific to svg, rereading some documents my case isn't entirely true anymore.

For an `a` element in different context it would return:
`<a href="#">Hello</a>`, tagName will return `A` and the function will return `a`
`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><a xlink:href="#" target="_top">World</a></svg>`, the tagName and function will return `a`

### Colors
For svg gradients you will get:
`<svg><linearGradient></linearGradient></svg>` tagName and the function will return `linearGradient`

What I previously thought (svg2 spec):
`<svg><solidcolor></solidcolor></svg>` tagName and the function will return `solidcolor`
But it changed to mimic the name conventions in previously used, so when it will be implemented it will be parsed as:
`<svg><solidColor></solidColor></svg>` tagName and the function willreturn `solidColor`

I prefer to use the exact name that is shown in the DOM, so an lowercase name for html elements and camelcased names for svg elements.

Also svg2 supports html elements inside an svg, I didn't bother to add the logic as at this moment in time there is no browser support, e.g. 
```svg
<svg xmlns="http://www.w3.org/2000/svg" xmlns:html="http://www.w3.org/1999/xhtml">
  <html:video src="http://example.org/dummyvideo" controls="controls">
    <html:p>The video format is not supported by this browser.</html:p>
  </html:video>
</svg>
```
So (currently) the P element returns an tagName of `html:p`, but the function should return `p`. 

***Conclusion**: The function could be removed now, as (I believe) this rule is the only rule that supports svg elements. An type of this function will be needed in the future. A better solution would probably be to check always the tagname with its namespaceURI together. So `a` elements within svg elements can never mess up the end results.

Reply:
Good point. I think you're right about the tagName + namespace approach. We shouldn't try to solve that here though. One problem at a time please :). I think we stick to the existing approach for this rule (e.g. `.tagName.toUpperCase()`) and create a separate issue / PR for this. Can you put together a new issue outlining what the problems are with the current way we're handling tag names?